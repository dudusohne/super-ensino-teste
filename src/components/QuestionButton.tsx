interface Button {
    name: string;
    click?: () => void;
}

export function QuestionButton({ name, click }: Button) {
    return (
        <button onClick={click} className="button-comp" style={{
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            borderRadius: '8px',
            border: '2px solid #38d0fc',
            maxWidth: '30rem',
            padding: '5px 38px 5px 38px',
            marginLeft: '1rem',
            marginTop: '5px',
            background: 'none'
        }}>
            <span style={{
                fontSize: '1rem',
                fontWeight: 'medium'
            }}>{name}</span>
        </button>
    )
}