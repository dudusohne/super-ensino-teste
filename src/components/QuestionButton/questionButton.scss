@import '../../styles/global.scss';

.button-comp {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  border-radius: 8px;
  border: 2px solid $primary;
  max-width: 30rem;
  padding: 5px 38px 5px 38px;
  margin-left: 1rem;
  margin-top: 5px;
  background: none;
  font-family: $font;
  font-weight: bold;
}

.button-comp:hover {
    background-color: $primary;
    cursor: pointer;
}
