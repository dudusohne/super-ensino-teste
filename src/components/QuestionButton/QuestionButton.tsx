import React from 'react'
import { Question } from '../../types';

import './questionButton.scss'

interface Button extends Question {
    click?: () => void;
    active?: boolean;
}

export function QuestionButton({ name, click, active }: Button) {
    return (
        <button onClick={click} className="button-comp" style={{ backgroundColor: `${!!active ? '#38d0fc' : '#fff'}`, }}>
            <span style={{ fontSize: '1rem', fontWeight: 'medium', color: '#313131' }}>{name}</span>
        </button>
    )
}