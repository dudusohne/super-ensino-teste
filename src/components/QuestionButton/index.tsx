import './questionButton.scss'

interface Button {
    name: string;
    click?: () => void;
}

export function QuestionButton({ name, click }: Button) {
    return (
        <button onClick={click} className="button-comp">
            <span style={{
                fontSize: '1rem',
                fontWeight: 'medium'
            }}>{name}</span>
        </button>
    )
}