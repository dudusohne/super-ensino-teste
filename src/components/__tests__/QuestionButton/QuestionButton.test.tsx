import renderer from 'react-test-renderer'
import { QuestionButton } from '../../QuestionButton'
import React from 'react'

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <QuestionButton name="asd" />,
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()

  // re-rendering
  tree = toJson(component)
  expect(tree).toMatchSnapshot()
})