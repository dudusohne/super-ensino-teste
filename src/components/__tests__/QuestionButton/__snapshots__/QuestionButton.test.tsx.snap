// Vitest Snapshot v1

exports[`Link changes the class when hovered 1`] = `
<button
  className="button-comp"
  style={
    {
      "backgroundColor": "#fff",
    }
  }
>
  <span
    style={
      {
        "color": "#313131",
        "fontSize": "1rem",
        "fontWeight": "medium",
      }
    }
  >
    asd
  </span>
</button>
`;

exports[`Link changes the class when hovered 2`] = `
<button
  className="button-comp"
  style={
    {
      "backgroundColor": "#fff",
    }
  }
>
  <span
    style={
      {
        "color": "#313131",
        "fontSize": "1rem",
        "fontWeight": "medium",
      }
    }
  >
    asd
  </span>
</button>
`;
